### PR TITLE
Storybook improvements / Design System

### PR DESCRIPTION
**Pull request by Lana-Lannister**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Contribution to [UI Components Library #44668](https://github.com/freeCodeCamp/freeCodeCamp/issues/44668)

# Improvement

- Added new accessibility [Addon a11y](https://storybook.js.org/addons/@storybook/addon-a11y) from Storybook;
- Added specific folder for components tests;
- Organized all components together for better Developer and Designer UX.

### CODE REVIEW

**Comment by Lana-Lannister** [comment]
Hi, my name is Lana! I'm a front-end software engineer at  @Pipefy. Before becoming a developer I was (still am) a FreeCodeCamp student ;) I'm very happy to start contributing and finding my way around the project. I'm building a Design System in my company and wish to contribute and learn with you all as well! 
If anyone wants to check it out, [here it is](https://pipefy.style/?path=/story/design-system-components-alert--alert-default)! 
cc: @ahmadabdolsaheb

**Comment by ahmadabdolsaheb** [comment]
Thanks for your interest in contributing to the UI component library.

We are moving forward to using storybook as a documentation tool for our design system down the road.
There are a few details we need to iron out; first of which is replacing react bootstrap with freeCodeCamp's  UI component library. 

You are welcome to pick up any of the bullet points there and add a new component to the library.

For this PR: Thanks for adding the a11y add on. let's move that dependency to dev-dependencies and this pr should be good to go.

The non-a11y add on modifications are not necessary for now.

**Comment by Lana-Lannister** [comment]
Done ;)  @ahmadabdolsaheb If it's okay, I can start contributing with the `modal` component. I'll open a new PR for that 😄

**Comment by ahmadabdolsaheb** [comment]
Awesome, just keep in mind that there might be a few pre-requisite components that need development first such as the button. 

for this pr: to clarify, everything other than the changes in tools/ui-components/.storybook/main.js , package.json,  package-lock.json can be removed from this pr.

We can discuss the storybook organization separately in a related issue if necessary.

File: package.json [hunk 1]
```diff
@@ -101,6 +101,7 @@
     "invariant": "2.2.4"
   },
   "devDependencies": {
+    "@storybook/addon-a11y": "^6.4.16",
```
Comment (huyenltnguyen) [inline package.json:104]: This addon should be added to the package.json file of the `ui-components` folder instead (https://github.com/freeCodeCamp/freeCodeCamp/blob/main/tools/ui-components/package.json). 

Additionally, the version should be fixed (without the `^`), I believe we have been following the [recommended practice](https://docs.renovatebot.com/dependency-pinning/#so-whats-best) and pin dev dependencies.

**Review by huyenltnguyen — COMMENTED** [review]
@Lana-Lannister Thank you for the PR!

As @ahmadabdolsaheb said, we are more than happy to have your help with the component library. The component library has not reached its maturity stage however, and we will need to move things around to keep the folder standardized.

I'll be helping with the development guide and tooling, as well as building some components, so let me know if you need help with anything.

**Comment by ahmadabdolsaheb** [comment]
@Lana-Lannister I went in and made the changes in the interest of time.
Looking forward to seeing your future contributions to the component library.

**Review by ahmaxed — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**